doc/dev/BUILD.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Building TileDB from source
+---
+
+## Prerequisites
+
+* CMake 3.21 or greater
+* C++ compiler with C++20 support
+    * The minimum compiler versions TileDB is being tested in CI are:
+        * MSVC from Visual Studio 2019 16.11
+        * GCC 10.3 on Linux and Windows via MinGW
+        * Apple Clang 14
+* Git (required by vcpkg)
+* curl (required by vcpkg on non-Windows)
+
+### Dependency management with vcpkg
```

Comment:
This section should be considered "Advanced Use" and appear later in the documentation. Nobody reading this document should have to read anything about `VCPKG_ROOT` to get started.

If a user does nothing special, `vcpkg` should just work. That's mostly what this section should say.

Reply:
I would actually like everybody who regularly builds the Core to set `VCPKG_ROOT`, but moved this section to the bottom of the document.